src/components/plan/PlanForm.svelte

```diff
@@ -139,6 +151,27 @@
         </Input>
       </Collapse>
     </fieldset>
+    <fieldset>
+      <Collapse title="Snapshots" padContent={false}>
+        <button class="st-button secondary" slot="right" on:click={onCreatePlanSnapshot}>Take Snapshot</button>
+        <div style="margin-top: 8px">
+          <CardList>
+            {#each $planSnapshotsWithSimulations as planSnapshot (planSnapshot.snapshot_id)}
+              <PlanSnapshot
+                activePlanSnapshotId={$planSnapshotId}
+                {planSnapshot}
+                on:click={() => setQueryParam(SearchParameters.SNAPSHOT_ID, `${planSnapshot.snapshot_id}`)}
```

Comment:
I had the tendency to press the back button to go back to the original plan after clicking on a snapshot to preview it, but `setQueryParam` just replaces the state in the history instead of pushing to it. Do you think we should allow the user to press the back button to go back to previous snapshots/plans?

Reply:
Seems like a reasonable behavior to support, can add that in.